packages/hop-node/src/watchers/BondTransferRootWatcher.ts

```diff
@@ -170,6 +179,11 @@ class BondTransferRootWatcher extends BaseWatcher {
       throw err
     }
   }
+
+  getAvailableCreditForBond (destinationChainId: number) {
+    const availableCredit = this.syncWatcher.getEffectiveAvailableCredit(destinationChainId)
```

Comment:
* Since the bond is occurring on L1, we need to get the available credit on L1
* The `effectiveAvailableCredit` returns `// L2 -> L1: (credit - debit - OruToL1PendingAmount - OruToAllUnbondedTransferRoots)`. For transfer roots, we just need `credit - debit`, like we do in original TransferRootWatcher [check](https://github.com/hop-protocol/hop/blob/33384d0da7f1e6de16ff80fc6cbc018067c91de3/packages/hop-node/src/watchers/BondTransferRootWatcher.ts#L138). With that said, it does not make sense to poll on-chain every 10s for both `credit` and `debit`. It would only make sense for this change if we expose a `this.rawAvailableCredit` (or something like that) that just returns an in-memory `credit - debit` that is polled in `syncBonderCredit` (or something like that). IIRC this change is a little heavier than I'd hope and is why I didn't just make this change w/o a PR. It looks like we already [call `getBaseAvailableCredit()`](https://github.com/hop-protocol/hop/blob/33384d0da7f1e6de16ff80fc6cbc018067c91de3/packages/hop-node/src/watchers/SyncWatcher.ts#L1136), which is what we need, in `calculateAvailableCredit()`, so maybe we can expose it there?
  * If that update is made, then I think it also makes sense to update the [original check](https://github.com/hop-protocol/hop/blob/33384d0da7f1e6de16ff80fc6cbc018067c91de3/packages/hop-node/src/watchers/BondTransferRootWatcher.ts#L138) to use this new in-memory check, just like we do with the BondWithdrawalWatcher ([getter lookup](https://github.com/hop-protocol/hop/blob/33384d0da7f1e6de16ff80fc6cbc018067c91de3/packages/hop-node/src/watchers/BondWithdrawalWatcher.ts#L75) is same as [check function lookup](https://github.com/hop-protocol/hop/blob/33384d0da7f1e6de16ff80fc6cbc018067c91de3/packages/hop-node/src/watchers/BondWithdrawalWatcher.ts#L144)).

Reply:
@shanefontaine Thanks for the clarification. I understand what you are saying about using the base available credit instead. Check out the [latest changes](https://github.com/hop-protocol/hop/pull/328/files) which cache `baseAvailableCredit` in SyncWatcher and the BondTransferRootWatcher uses now. I think this correct but let me know